File: show-duplicates/display_duplicates.go
package main

import "fmt"

func displayDuplicates(nums []int) int {
	duplicates := []int{} // store duplicates
	seen := map[int]bool{} // track seen numbers

	for _, num := range nums {
		if seen[num] {
			if !contains(duplicates, num) {
				duplicates = append(duplicates, num)
			}
		} else {
			seen[num] = true
		}
	}
	return len(duplicates)
}

func contains(slice []int, num int) bool {
	for _, v := range slice {
		if v == num {
			return true
		}
	}
	return false
}

func main()  {
	nums := []int{10, 20, 60, 30, 20, 40, 30, 60, 70, 80}
	fmt.Println(displayDuplicates(nums))
}

File: show-duplicates/display_duplicates.ts


const displayDuplicates = (nums: number[]): number => {
    // create array to store duplicates
    let duplicates: number[] = []
    let seen: number[] = []

    // loop through numbers
    // add new number to the duplicates array
    // if it's already there, skip it
    for (let i = 0; i < nums.length; i++) {
        if (seen.includes(nums[i]) && !duplicates.includes(nums[i])) {
            duplicates.push(nums[i])
        } else {
            seen.push(nums[i])
        }
    }

    //return the length of the duplicates array
    return duplicates.length
}

console.log(displayDuplicates([10, 20, 60, 30, 20, 40, 30, 60, 70, 80]))